main.go

```diff
@@ -138,6 +139,9 @@ func parseFlags() *config {
 	flag.BoolVar(&cfg.usePathsCache, "use-paths-cache", false,
 		"Use a cache to store metrics paths lists.",
 	)
+	flag.BoolVar(&cfg.ignoreReadError, "ignore-read-error", false,
+		"Ignore all read errors to avoid breaking prometheus",
```

Comment:
Ignore all read errors and return empty results instead. When eabled prometheus will display only local points instead of returning an error.